src/main/scala/mimir/adaptive/CheckHeader.scala

```diff
@@ -0,0 +1,341 @@
+package mimir.adaptive
+
+import java.io._
+import com.typesafe.scalalogging.slf4j.LazyLogging
+import scala.collection.JavaConverters._
+
+import mimir.Database
+import mimir.algebra._
+import mimir.lenses._
+import mimir.models._
+import mimir.statistics.FuncDep
+
+object CheckHeader
+  extends Multilens
+  with LazyLogging
+{
+  def checkheader(db: Database, config: MultilensConfig): Boolean = {
+    val col = config.query.expressions(0);
+    val tablename = col.toString.split("_")(0);
+    var detect  = detect_header(db, tablename);
+
+    var len = 0;
+    var arrs : Seq[mimir.algebra.PrimitiveValue] = null
+
+    var str=""
+    db.query("SELECT * FROM " + tablename + "_RAW limit 1 ;")(_.foreach{result =>
+      arrs =  result.tuple
+    })
+    len = arrs.length
+
+    for(i<- 0 until len){
+      val res = arrs(i)
+      var ch =  res.toString()(1)
+      if(ch.toByte >= '0' && ch.toByte <= '9'){
+        str ++= s"""COLUMN_$i AS COL_$i ,""";
+      }
+      else{
+        str ++= s"""COLUMN_$i AS $res ,""";
+      }
+    }
+
+    var query = ""
+
+    str = str.slice(0,(str.length()-2));
+    str = str.replaceAll("\\'","");
+    query = "CREATE VIEW "+ tablename +"_Header AS SELECT " +str +" from "+ tablename + "_RAW limit 1,1000000000000"
+    db.backend.update(query)
+    return detect;
+  }
+
+  def detect_header(db: Database, table: String): Boolean = {
```

Comment:
detect_header() belongs in a Model object.  Please review https://github.com/UBOdin/mimir/wiki/Concepts-Models